log/logger.go

```diff
@@ -1,13 +1,31 @@
 package log
 
 import (
+	"encoding"
+	"encoding/json"
+	"fmt"
 	"io"
 
 	"github.com/pkg/errors"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/pkgerrors"
 )
 
+func init() {
+	zerolog.InterfaceMarshalFunc = func(i interface{}) ([]byte, error) {
+		switch v := i.(type) {
+		case json.Marshaler:
+			return json.Marshal(i)
+		case encoding.TextMarshaler:
+			return json.Marshal(i)
+		case fmt.Stringer:
+			return fmt.Appendf([]byte("\""), "%s%s", v.String(), "\""), nil
```

Comment:
Could you verify if we need to escape characters form `.String()`? If it contains a `"` it may break the json output.

Reply:
yes, I was also confused at the beginning, I have verified that `zerolog` does the same for [string](https://github.com/rs/zerolog/blob/master/internal/json/string.go#L44) type